var/spack/repos/builtin/packages/dd4hep/package.py

```diff
@@ -158,13 +166,18 @@ class Dd4hep(CMakePackage):
     depends_on("geant4@10.2.2:", when="+ddg4")
     depends_on("assimp@5.0.2:", when="+ddcad")
     depends_on("hepmc3", when="+hepmc3")
+    depends_on("hepmc3@3.2.6:", when="+hepmc3-gz")
+    depends_on("bzip2", when="+hepmc3-gz")
+    depends_on("lzma", when="+hepmc3-gz")
```

Comment:
Better to depend on `xz` since even the `lzma` project [directs](https://tukaani.org/lzma/) users there. 
```suggestion
    depends_on("xz", when="+hepmc3-gz")
```